### PR TITLE
support SPOT for secondary worker.

### PR DESCRIPTION
**Pull request by duqDarren**
Support 'SPOT' under preemptibility for secondary worker.

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
dataproc: added support for `SPOT` option for `preemptibility` in `google_dataproc_cluster`
```

### CODE REVIEW

**Comment by melinath** [comment]
we will not be able to merge this as-is because it adds the field to the GA provider.

**Comment by melinath** [comment]
yep

**Comment by duqDarren** [comment]
@trodge @melinath 
https://github.com/GoogleCloudPlatform/magic-modules/pull/7002 deprecated the v1beta2. 
I also removed the version guard from this PR.

**Comment by melinath** [comment]
Great! @trodge feel free to assign me as a secondary reviewer when you're done.

**Review by trodge — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb [hunk 1]
```diff
@@ -680,7 +680,7 @@ func resourceDataprocCluster() *schema.Resource {
 											"cluster_config.0.preemptible_worker_config.0.disk_config",
 										},
 										ForceNew:     true,
-										ValidateFunc: validation.StringInSlice([]string{"PREEMPTIBILITY_UNSPECIFIED", "NON_PREEMPTIBLE", "PREEMPTIBLE"}, false),
+										ValidateFunc: validation.StringInSlice([]string{"PREEMPTIBILITY_UNSPECIFIED", "NON_PREEMPTIBLE", "PREEMPTIBLE", "SPOT"}, false),
```
[the remark this answers is not in the document]
Comment (duqDarren) [reply mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb]: Oh, right, just removed it.

**Review by trodge — COMMENTED** [review]
LGTM, added @melinath as a secondary reviewer.

**Review by melinath — APPROVED** [review]
This seems extremely straightforward.

**Pull request merged**